pkg/connector/builder.go

```diff
@@ -72,10 +73,13 @@ func (b *DefaultBuilder) Init(c Connector, id string, config Config) error {
 		return cerrors.Errorf("could not create plugin %q: %w", config.Plugin, err)
 	}
 
+	tn := time.Now()
```

Comment:
This has to be moved to `Build`, otherwise we'll have issues when retrieving connectors from the store ([this function](https://github.com/ConduitIO/conduit/blob/7c852bff2541a1faee9b688e75f40d72c74e3a35/pkg/connector/store.go#L166)).

The store first calls `Build` to get the connector of the right type, then unmarshals the stored value and then calls `Init`. The stored value will contain `UpdatedAt` and `CreatedAt` so we should use those values and not create new ones in `Init`.